fe/fe-core/src/main/java/org/apache/doris/planner/HashJoinNode.java

```diff
@@ -140,6 +141,60 @@ public HashJoinNode(PlanNodeId id, PlanNode outer, PlanNode inner, TableRef inne
         }
     }
 
+    public HashJoinNode(PlanNodeId id, PlanNode outer, PlanNode inner, JoinOperator joinOp, List<Expr> eqJoinConjuncts,
```

Comment:
Why this PR include `planner`?

Reply:
> Why this PR include `planner`?

And any other modification to Planner is based on the similiar reason as commented above